src/resolvers/Mutation/createPost.ts

```diff
@@ -93,6 +95,47 @@ export const createPost: MutationResolvers["createPost"] = async (
     imageUrl: args.file ? uploadImageObj?.newImagePath : "",
   });
 
+  // Check if the pinned property is set
+  if (args.data.pinned) {
+    // Check if the user has privileges to pin the post
+    const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
+      (organizationId) => organizationId.toString() === args.data.organizationId
+    );
+
+    if (
```

Comment:
I don't think you should throw error when you've already created the post. Do it before creating the post. 

Or don't throw the error and instead, create the post but change the value of input `pinned` boolean to false, if the user who sent it is a normal user.

Reply:
In the latest push, I am checking to see if the user is authorized to create the same or not. If not, Ill throw the error, otherwise I continue to make the post after that.